core/arch/interp.c

```diff
@@ -3377,13 +3381,15 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             bb->instr_start = bb->cur_pc;
             if (bb->full_decode) {
                 /* only going through this do loop once! */
-                bb->cur_pc = decode(dcontext, bb->cur_pc, bb->instr);
+                bb->cur_pc = IF_AARCH64_ELSE(decode_with_ldstex, decode)
+                    (dcontext, bb->cur_pc, bb->instr);
                 if (bb->record_translation)
                     instr_set_translation(bb->instr, bb->instr_start);
             } else {
                 /* must reset, may go through loop multiple times */
                 instr_reset(dcontext, bb->instr);
-                bb->cur_pc = decode_cti(dcontext, bb->cur_pc, bb->instr);
+                bb->cur_pc = IF_AARCH64_ELSE(decode_cti_with_ldstex, decode_cti)
```

Comment:
Oh, I see.  Not really "straight-line" or "contiguous", as it can contain multiple jumps (conditional or not) and skip over other code, but "within a short bounded area" of 64 bytes.  Yes, l see that the bundling code does that.  OK, that mitigates some of the issues, though not all, and at the cost of not handling all ldxr..strxr sequences.